### PR TITLE
docs(links): fix outdated links and imrpove formatting

### PR DESCRIPTION
**Pull request by Sh3Rm4n**
### Description of the feature or fix

While reading through the documentation, I've noticed some links were not working, and I started to look for other links not working anymore as well. Additionally, I've fixed some small formatting issues. 

### Checkpoints

- [ ] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [x] Update the documentation

### CODE REVIEW

File: docs/libs/freetype.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ You can simply pass the path to the font as usual on your operating system or pl
 
 ## Learn more
 - FreeType [tutorial](https://www.freetype.org/freetype2/docs/tutorial/step1.html)
-- LVGL's [font interface](https://docs.lvgl.io/v7/en/html/overview/font.html#add-a-new-font-engine)
+- LVGL's [font interface](https://docs.lvgl.io/master/overview/font.html?highlight=font#add-a-new-font-engine)
```
Comment (embeddedt) [inline docs/libs/freetype.md]:
```suggestion
- LVGL's [font interface](https://docs.lvgl.io/master/overview/font.html#add-a-new-font-engine)
```

**Review by embeddedt — COMMENTED** [review]
One minor suggestion.

File: docs/libs/freetype.md (hunk 1 above)
Comment (Sh3Rm4n) [reply docs/libs/freetype.md]: Thanks :)

**Review by kisvegabor — APPROVED** [review]
LGTM, thank you!

**Pull request merged**